backend/tournesol/tests/test_suggestions.py

```diff
@@ -399,6 +434,30 @@ def test_suggestions_for_new_user(self):
         suggestions = suggester.get_second_video_recommendation(new_user, self.videos[9].uid, 6)
         assert len(suggestions) == 6
 
+    def test_sparsification_metric(self):
+        suggester = SuggestionProvider(self.poll)
+        suggester.get_first_video_recommendation(self.sparsity_comparison_user, 6)
+        user_graph = suggester._user_specific_graphs[self.sparsity_comparison_user.id]
+
+        for n in user_graph.nodes:
+            if n.uid == self.videos[0].uid:
+                alone_node = n
+                break
+
+        for n in user_graph.nodes:
+            max_value = n.graph_sparsity(alone_node)
+            for m in user_graph.nodes:
+                assert n.graph_sparsity(m) <= max_value
```

Comment:
What's the intention here again? It seems that in this particular setup, all graph sparsity values are 1.
So this could be simplified as
```suggestion
        for n in user_graph.nodes:
            for m in user_graph.nodes:
                assert n.graph_sparsity(m) == 1
```

But is it really the expected result?

Reply:
Actually no, the values should be all < 1 and especially low for all nodes except when these are compared with the one containing video 0 (this is due to the comparison graph, all nodes are densely connected but this one)
If all are equal to 1 I may have inverted a comparison

Reply:
I checked, there is a problem with the normalization applied, in fact there should be a symmetric normalized adjacency matrix, I look to fix it

Reply:
Ok, it looks way better now :)